fhirstore/fhirstore.py

```diff
@@ -290,75 +290,72 @@ def search(
         """
         self.validate_resource_type(resource_type)
 
-        core_query = build_core_query(params)
-        query = {
-            "min_score": 0.01,
-            "from": offset,
-            "size": result_size,
-            "query": core_query,
-        }
-
-        if sort:
-            query["sort"] = sort
-
-
-        if elements:
-            query["_source"] = elements
-
-        # .lower() is used to fix the fact that monstache changes resourceTypes to
-        # all lower case
-        hits = self.es.search(body=query, index=f"fhirstore.{resource_type.lower()}")
-        bundle = {
-            "resource_type": "Bundle",
-            "items": [
-                {"resource": h["_source"], "search": {"mode": "match"}}
-                for h in hits["hits"]["hits"]
-            ],
-
-            "total": hits["hits"]["total"]["value"],
-        }
-
-        if elements:
-            bundle["tag"] = {"code": "SUBSETTED"}
-
-        if include:
-            #For each result instance
-            for item in bundle["items"]:
-                #For each attribute to include
-                for attribute in include:
-                    # split the reference attribute "Practioner/123" into a 
-                    # resource "Practioner" and an id "123"
-                    included_resource, included_id = re.split(
-                        "\/", item[attribute]["reference"], maxsplit=1
-                    )
-                    # search the db for the specific resource to include
-                    included_hits = self.es.search(
-                        body={"simple_query_string": {"query": included_id, "fields": "id",}},
-                        index=f"fhirstore.{included_resource.lower()}",
-                    )
-            [
-                bundle["items"].append({"resource": h["_source"], "search": {"mode": "include"}})
-                for h in included_hits["hits"]["hits"]
-            ]
+        raw_args = SearchArguments(args, resource_type)
 
-        return bundle
+        parsed_args = self.urlparser.parse_arguments(raw_args)
+        core_query = self.corequerybuilder.build_core_query(parsed_args.core_args)
```

Comment:
I think you should have to give parsed_args.core_args as argument here but only something like parsed_args
Maybe self.urlparser.parse_arguments could return the core_args directly?

Reply:
not sure I agree here, `urlparser.parse_arguments` returns core_args and formatting_args (maybe the function could return these 2 values, one would be passed to `corequerybuilder.build_core_query` and the other one to ` formatter.initiate_bundle`. However I agree that `parsed_args` is not very clear since we cannot know what it is based on just its name